File: Backend/movie-booking-app/src/main/java/com/booking/service/TicketService.java
package com.booking.service;

import java.util.List;

import com.booking.exception.SeatsAreNotAvailabeExceptions;
import com.booking.model.Ticket;
import com.booking.model.TicketDto;

public interface TicketService {

	TicketDto bookTicket(TicketDto ticketDto) throws SeatsAreNotAvailabeExceptions;

	List<Ticket> getAllTickets(int movieId);

}

File: Frontend/src/app/services/ticket.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { TicketDto } from '../model/ticket-dto';

@Injectable({
  providedIn: 'root'
})
export class TicketService {

  baseUrl: string = 'http://localhost:9090/api/v1/moviebooking/ticket'

  constructor(private httpClient: HttpClient) { }

  public bookTicket(ticketDto: TicketDto) {
    return this.httpClient.post<any>(this.baseUrl + '/book', ticketDto);
  }

  public getTicketsByMovieId(movieId: number) {
    return this.httpClient.get(this.baseUrl + '/movie/' + movieId)
  }

}
